plot/plot.app.js

```diff
@@ -684,13 +690,42 @@
+                                                                    Title: traceLabel,
+                                                                    Context_Params: qCharacter+"pcid="+contextUrlParams.cid+"&ppid="+contextUrlParams.pid
+                                                                };
+                                                                traceLabel=ERMrest.renderMarkdown(ERMrest.renderHandlebarsTemplate(trace.legend_markdown_pattern,$rootScope.templateParams),true);
+                                                                var matchs =/<a\b.+href="([^\n\r]*)"\s/ig.exec(traceLabel);
+                                                                values.legend_markdown_pattern.push(matchs[1]);
+                                                                delete $rootScope.templateParams;
+                                                            }
+                                                            values.labels.push(traceLabel);
+                                                            values.text.push(row[trace.legend_col]);
+                                                            console.log(values.text)
                                                         }
                                                         values.values.push(formatData(row[trace.data_col], plot.config ? plot.config.format_data : false, "pie"));
+                                                    
                                                     });
                                                     plot_values.data.push(values);
+                                                    if(plot.config.hasOwnProperty("title_hyperlink_pattern")){
+                                                        var qCharacter = /\((.*?)\)/ig.exec(plot.config.title_hyperlink_pattern)[1].indexOf("?") !== -1 ? "&" : "?";
+                                                        console.log(qCharacter)
+                                                        $rootScope.templateParams= {
+                                                            Title: layout.title,
+                                                            Context_Params: qCharacter+"pcid="+contextUrlParams.cid+"&ppid="+contextUrlParams.pid
+                                                        };
+                                                        layout.title=ERMrest.renderMarkdown(ERMrest.renderHandlebarsTemplate(plot.config.title_hyperlink_pattern,$rootScope.templateParams),true);
+                                                        delete  $rootScope.templateParams;
+                                                    }
                                                     plot_values.layout = layout;
                                                     plot_values.config = config;
+                                                    plot_values.layout.disable_legend_click=plot.config.disable_legend_click;
```

Comment:
Adding the disable_legend_click flag to configuration(Will rename it to disable_default_legend_click in next PR)